debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java

```diff
@@ -312,7 +301,6 @@ private void stop(boolean restart) {
                 setTaskState(State.RESTARTING);
                 if (restartDelay == null) {
                     restartDelay = ElapsedTimeStrategy.constant(Clock.system(), retriableRestartWait.toMillis());
-                    restartDelay.hasElapsed();
```

Comment:
IMHO by removing it you ae removing the pause.

Reply:
Right, but only the first one pause:-) [This method](https://github.com/debezium/debezium/blob/main/debezium-core/src/main/java/io/debezium/util/ElapsedTimeStrategy.java#L50) says, if the delay interval has already elapsed or not. As a side effect it recomputes next timestamp for waiting. This method is very confusing (it's description, side effects and initialization) and IMHO should be also refactored.

However, you are right that removing it is wrong - due to its side effect if this call is removed, the elapsed interval won't be initialized and actually won't do any wait when the strategy is used for the first time. So basically this has to be called every time right after `ElapsedTimeStrategy` is created.

Reply:
@vjuranek I agree on the refactoring